### PR TITLE
escape URIs on output

### PR DESCRIPTION
**Pull request by humanchimp**
~~DO NOT MERGE YET~~

This PR aims to escape URIs on output using a method that will not re-escape already escaped parts of the text, making it safe to use without running the risk of double-escaping the text.

Some examples:

| Input | Output |
|-------------------|--------------|
|`https://www.getlytics.com/hello  world` |`https://www.getlytics.com/hello%20%20world`|
|`https://www.getlytics.com/hello %20world` |`https://www.getlytics.com/hello%20%20world`|
|`https://www.getlytics.com/hello%20 world`|`https://www.getlytics.com/hello%20%20world`|
|`https://www.getlytics.com/hello%20%20world`|`https://www.getlytics.com/hello%20%20world`|

Opening for discussion. I still need to get the tests passing/add some of my own.

### CODE REVIEW

**Comment by markhayden** [comment]
👍 
in the description you have input `https://www.getlytics.com/hello world` and output `https://www.getlytics.com/hello%20%20world` but the output should be / is `https://www.getlytics.com/hello%20world` super nit but might be worth fixing just because

thanks for adding this

**Comment by humanchimp** [comment]
there are actually 2 spaces there, but it's hard to see.

File: dist/pathfora.js [hunk 1]
```diff
@@ -562,7 +626,7 @@
      */
     parseQuery: function (url) {
       var query = {};
-      var pieces = url.split('?');
+      var pieces = utils.escapeURI(url, { keepEscaped: true }).split('?');
```
Comment (sarahyablok) [inline dist/pathfora.js]: By calling this function here, I believe we'd only be escaping the queries. Thus:

- `http://localhost:8080/examples/_bar.html?hello=hi there`
would match
- `http://localhost:8080/examples/_bar.html?hello=hi%20there`

but: 
- `http://localhost:8080/examples/_bar hi.html`
would not match 
- `http://localhost:8080/examples/_bar%20hi.html`

I might be wrong, but it seems like the desired functionality is to escape the whole url.

Comment (humanchimp) [inline dist/pathfora.js]: ok, let me look into that

Comment (humanchimp) [inline dist/pathfora.js]: Yeah, you were totally right, of course. I updated the PR. I think I got it right this time.

**Comment by sarahyablok** [comment]
👍

**Comment by humanchimp** [comment]
I rebased this on develop and resolved the conflict about the trailing slash. Gonna push the button once green.

**Pull request merged**